### PR TITLE
[microshift] Add initial support for backup info

### PR DESCRIPTION
**Pull request by pacevedom**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/microshift.py [hunk 1]
```diff
@@ -26,7 +27,7 @@ class Microshift(Plugin, RedHatPlugin):
     short_desc = 'Microshift'
     plugin_name = 'microshift'
     plugin_timeout = 900
-    packages = ('microshift', 'microshift-selinux', 'microshift-networking')
+    packages = ('microshift', 'microshift-selinux', 'microshift-networking', 'microshift-greenboot')
```
Comment (TurboTurtle) [inline sos/report/plugins/microshift.py:30]: Please split this into multiple lines to adhere to PEP8.

File: sos/report/plugins/microshift.py [hunk 2]
```diff
@@ -146,7 +147,13 @@ def setup(self):
         Output format for this function is based on `oc adm inspect` command,
         which is used to retrieve all API resources from the cluster.
         """
-        self.add_forbidden_path('/var/lib/microshift')
+        self.add_service_status('greenboot-healthcheck')
+        self.add_journal('greenboot-healthcheck')
```
Comment (TurboTurtle) [inline sos/report/plugins/microshift.py:151]: If you add this service to the `services` tuple (akin to `packages`, `commands`, etc...), these collections become implicit.

File: sos/report/plugins/microshift.py [hunk 3]
```diff
@@ -146,7 +147,13 @@ def setup(self):
...
+
+        if os.path.exists('/var/lib/microshift-backups'):
```
Comment (TurboTurtle) [inline sos/report/plugins/microshift.py:153]: Use Plugin's `self.path_exists()` here, as this is a smaller wrapper to account for when we're running in a container and need to collect from the host and not the container filesystem.

Comment (pacevedom) [reply sos/report/plugins/microshift.py]: Done.

File: sos/report/plugins/microshift.py (hunk 1 above)
Comment (pacevedom) [reply sos/report/plugins/microshift.py]: Done.

File: sos/report/plugins/microshift.py (hunk 2 above)
Comment (pacevedom) [reply sos/report/plugins/microshift.py]: Done.

File: sos/report/plugins/microshift.py [hunk 4]
```diff
@@ -146,7 +148,10 @@ def setup(self):
         Output format for this function is based on `oc adm inspect` command,
         which is used to retrieve all API resources from the cluster.
         """
-        self.add_forbidden_path('/var/lib/microshift')
+        if self.path_exists('/var/lib/microshift-backups'):
+            self.add_copy_spec(['/var/lib/microshift-backups/*/version',
+                                '/var/lib/microshift-backups/health.json'])
```
Comment (dhellmann) [inline sos/report/plugins/microshift.py:153]: I think @pmtk changed the name of this file.

**Review by TurboTurtle — APPROVED** [review]
Ack, pending confirmation of the file name mentioned above.

**Comment by dhellmann** [comment]
I like the use of the wildcard, that will mean fewer updates here if we add more metadata files.

This looks good to me.

**Pull request merged**